virtcontainers/kata_agent.go

```diff
@@ -429,6 +408,22 @@ func (k *kataAgent) startProxy(sandbox *Sandbox) error {
 		return nil
 	}
 
+	if s, ok := k.vmSocket.(kataVSOCK); ok {
+		data, err := sandbox.hypervisor.hotplugAddDevice(nil, vSockDev)
+		if err != nil {
+			return err
+		}
+
+		contextID, ok := data.(uint32)
+		if !ok {
+			return fmt.Errorf("failed to get guest context ID")
+		}
+		sandbox.Logger().Info("Got context ID ", contextID)
+		s.contextID = contextID
+		s.port = uint32(vSockPort)
```

Comment:
Both `port` and `contextID` can be written directly in the `kataAgent` handler:
```go
s.contextID = contextID
s.port = uint32(vSockPort)
```
replaced by
```go
k.vmSocket.contextID = contextID
k.vmSocket.port = uint32(vSockPort)
```
and remove the assignment 
```go
k.vmSocket = s
```

Reply:
I don't think so, `vmSocket` is an interface  https://github.com/kata-containers/runtime/blob/master/virtcontainers/kata_agent.go#L94